src/main/java/net/javadiscord/javabot/listener/CodeAttachmentListener.java

```diff
@@ -0,0 +1,123 @@
+package net.javadiscord.javabot.listener;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Listens for messages containing "message.txt" or similar file attachments
+ * containing source code, in order to format the code into one or more code
+ * snippet blocks in the Discord channel instead. The code blocks are sent in
+ * a series of messages that chain their replies to the original message.
+ */
+public class CodeAttachmentListener extends ListenerAdapter {
+	@Override
+	public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+		if (isCodeMessage(event.getMessage())) {
+			fetchSource(event.getMessage())
+				.thenApply(this::processSource)
+				.thenAcceptAsync(blocks -> {
+					Message lastMsg = event.getMessage();
+					for (var block : blocks) {
+						lastMsg = lastMsg.reply(block).complete();
+					}
+				});
+		}
+	}
+
+	/**
+	 * Checks if a message is considered to be a valid "code" message that we
+	 * can process into code blocks. A message is a code message if it contains
+	 * exactly 1 attachment whose name is "message.txt" and whose size does not
+	 * exceed 10KB.
+	 * @param msg The message to check.
+	 * @return True if the message is eligible for processing, or false otherwise.
+	 */
+	private boolean isCodeMessage(Message msg) {
+		return msg.getAttachments().size() == 1 &&
+				msg.getAttachments().get(0).getFileName().equals("message.txt") &&
+				msg.getAttachments().get(0).getSize() < 10000;
+	}
+
+	/**
+	 * Fetches the source code attached to a message.
+	 * @param msg The message to fetch from.
+	 * @return A future that completes when the source code is fetched.
+	 */
+	private CompletableFuture<String> fetchSource(Message msg) {
+		return msg.getAttachments().get(0).retrieveInputStream().thenApply(in -> {
+			try {
+				String s = new String(in.readAllBytes());
+				in.close();
```

Comment:
this should be done either in a `finally`-block or using a `try-with-resources` statement. Such a `try-with-resources` block can be created like this if the variable is declared earlier:
```java
try(InputStream is=in){
	  //code using the InputStream
}
```

Reply:
Done